test/strings/basic.jl

```diff
@@ -1409,3 +1409,18 @@ end
         @test transcode(String, transcode(UInt8, transcode(UInt16, str))) == str
     end
 end
+
+@testset "cwstring" begin
```

Comment:
`Base.cwstring` is only defined on Windows: https://github.com/JuliaLang/julia/blob/b0c1525f1731186767ae42e7d625bf0909e49af8/base/strings/cstring.jl#L108-L124 This testset should only be run on Windows, otherwise it'll fail on other platforms, which means you should change this to
```julia
if Sys.iswindows()
    @testset "cwstring" begin
        # ...
    end
end
```

Reply:
Done. Testset still fails (intentionally) on Windows, will my branch still be merged or should a fix addressing the tests be pushed first?

Reply:
We aren't going to merge a PR with broken tests (unless [explicitly marked so](https://docs.julialang.org/en/v1/stdlib/Test/#Test.@test)), but I'm not clear why we should just add a broken test.  What's broken?  The test?  The code?

Reply:
Base.cwstring() converts an input string s to a NUL-terminated character vector but currently throws an error for any input string s that contains a NUL character anywhere. 
```julia
str_2 = "Wordu\000"
# ...
@test Base.cwstring(str_2) == UInt16[0x0057, 0x006f, 0x0072, 0x0064, 0x0000]
```
I input a string with a terminating NUL expecting cwstring() to trim the NUL off the end of the input and proceed to convert the remainder as normal, but instead cwstring() errors. Sorry for not clarifying earlier, but my ultimate question is should we add terminal NUL trimming to cwstring() or just leave the function alone? If the latter, I can just change my broken test to
```julia
@test_throws ArgumentError Base.cwstring(str_2)
```
so all tests in the testset work and we can merge this PR without issue.